### PR TITLE
Update declaration

### PR DESCRIPTION
**Pull request by ktsn**
Related to #215 

I moved all type-related files to `types` directory.
And I added test cases for the d.ts file. You can run the test cases by executing the following command. (It's ok if there are no error is printed)

```sh
$ cd types
$ typings install # install d.ts of Vue.js
$ tsc -p test/tsconfig.json 
```

### CODE REVIEW

**Comment by yyx990803** [comment]
Looks great - in the meanwhile, I'm trying to merge `next` branch with `master` for Vuex 1.0 which will support both Vue 1.x and 2.0. Would you mind rebasing after that?

**Comment by ktsn** [comment]
Sure! I will rebase that.

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
+declare namespace Vuex {
+  class Store<S> {
+    constructor(options: StoreOption<S>);
+
+    state: S;
+
+    dispatch(mutationName: string, ...args: any[]): void;
+    dispatch<P>(mutation: MutationObject<P>): void;
+
+    watch(path: string, cb: (value: any) => void, options?: WatchOption): void;
```
Comment (yyx990803) [inline types/index.d.ts]: I've merged `next` into `master` and this is the only thing that is deprecated. LGTM after removing this.

**Comment by ktsn** [comment]
I've removed it :)

**Comment by yyx990803** [comment]
Thanks! Sorry for the back and forth, but I just pushed some additional changes for 1.0 today. In brief:

- `middlewares` changed to `plugins`. (#233)
- new method: `replaceState(state)`
- new method: `on`, `once`, `off` and `emit`.

Also see (the latest api reference)[https://github.com/vuejs/vuex/blob/master/docs/en/api.md].

**Comment by ktsn** [comment]
OK, I'll add the types for them a little later!

**Comment by ktsn** [comment]
I'm done.

**Comment by yyx990803** [comment]
Great work, thank you very much! Do we need to do anything to publish this to typings?

**Comment by ktsn** [comment]
We need to send PR to [typings/registry](https://github.com/typings/registry) with a json file like [this](https://github.com/typings/registry/blob/master/npm/debug.json).
If you don't mind, I will do that too.

**Comment by yyx990803** [comment]
@ktsn sure!

**Pull request merged**